.paw/work/agent-installation-management/ImplementationPlan.md

```diff
@@ -0,0 +1,1141 @@
+- Automatic Copilot refresh or reload after installation
+- Dynamic agent discovery from external registries
+- Rollback mechanisms to restore previous agent versions
+- Telemetry or analytics about installation success rates
+- Cross-machine synchronization of custom instruction files
+- GUI for managing or viewing installed agents
+- Hot-reload of agent changes without VS Code restart
+
+## Implementation Approach
+
+The implementation follows a three-phase approach:
+
+1. **Core Infrastructure**: Add platform detection, path resolution, and agent template bundling to provide foundation for installation
+2. **Installation Logic**: Implement agent composition, file writing, version tracking, and activation event handling
+3. **Update and Migration**: Add upgrade detection, migration manifest, and cleanup of obsolete files
+
+Each phase builds incrementally and includes comprehensive testing at both automated and manual levels. The approach prioritizes reliability through idempotent operations, graceful error handling, and detailed logging.
+
+---
+
+## Phase 1: Core Infrastructure
+
+### Overview
+This phase establishes the foundational capabilities needed for agent installation: detecting the platform and VS Code variant, resolving the correct prompts directory path, and bundling agent templates with the extension.
+
+### Changes Required:
+
+#### 1. Platform Detection Module
+**File**: `vscode-extension/src/agents/platformDetection.ts` (new)
+**Changes**: Create module to detect operating system and VS Code distribution variant
+
+```typescript
+import * as os from 'os';
+import * as path from 'path';
+
+/**
+ * Supported VS Code distribution variants
+ */
+export enum VSCodeVariant {
+  Code = 'Code',
+  Insiders = 'Code - Insiders',
+  CodeOSS = 'Code - OSS',
+  VSCodium = 'VSCodium'
+}
+
+/**
+ * Platform information for determining paths
+ */
+export interface PlatformInfo {
+  platform: NodeJS.Platform;
+  homeDir: string;
+  variant: VSCodeVariant;
+}
+
+/**
+ * Get current platform information
+ */
+export function getPlatformInfo(): PlatformInfo {
+  return {
+    platform: os.platform(),
+    homeDir: os.homedir(),
+    variant: detectVSCodeVariant()
+  };
+}
+
+/**
+ * Detect which VS Code variant is running
+ * Detection based on environment variables and process information
+ */
+export function detectVSCodeVariant(): VSCodeVariant {
+  // Check environment variable set by VS Code variants
+  const productName = process.env.VSCODE_PRODUCT_NAME;
+  
+  if (productName?.includes('Insiders')) {
+    return VSCodeVariant.Insiders;
+  }
+  if (productName?.includes('OSS')) {
+    return VSCodeVariant.CodeOSS;
+  }
+  if (productName?.includes('VSCodium')) {
+    return VSCodeVariant.VSCodium;
+  }
+  
+  // Default to standard Code
+  return VSCodeVariant.Code;
+}
+
+/**
+ * Resolve the User/prompts directory for current platform and VS Code variant
+ */
+export function resolvePromptsDirectory(info: PlatformInfo): string {
```

Comment:
Add a setting that can allow the user to set the prompt directory. This is default unset, and if it's unset the logic here will be used to attempt to find it. If it's an unsupported platform, the error message should indicate that the setting can be set to locate the prompt directory. If that value is set, it will override this logic. In this way, if hits a situation where the prompt directory is not in the place we expect, there is a setting that can unblock them. Update the plan and also the Spec if necessary.